skfem/utils.py

```diff
@@ -597,6 +603,68 @@ def condense(A: spmatrix,
     return ret_value if len(ret_value) > 1 else ret_value[0]
 
 
+def mpc(A: spmatrix,
+        b: ndarray,
+        S: Optional[DofsCollection] = None,
+        M: Optional[DofsCollection] = None,
+        T: Optional[spmatrix] = None,
+        g: Optional[ndarray] = None) -> CondensedSystem:
+    """Apply a multipoint constraint on the linear system.
+
+    Parameters
+    ----------
+    A
+    b
+        The linear system to constrain.
+    S
+        The set of DOFs that are eliminated from the system.
+    M
+        The set of DOFs that are kept in the system.
+    T
+    g
+        The constraint is of the form `x[M] = T @ x[S] + g`.
+
+    """
+    if M is None:
+        M = np.array([], dtype=np.int64)
+    else:
+        M = _flatten_dofs(M)
+    if S is None:
+        raise NotImplementedError("S must be specified.")
+    S = _flatten_dofs(S)
+
+    assert isinstance(S, ndarray) and isinstance(M, ndarray)
+    U = np.setdiff1d(np.arange(A.shape[0]), np.concatenate((M, S)))
+
+    if T is None:
+        T = sp.eye(len(S), len(M))
+    if g is None:
+        g = np.zeros(len(S))
+
+    B = bmat([
+        [
+            A[U].T[U].T,
+            A[U].T[M].T + A[U].T[S].T @ T,
+        ],
+        [
+            T.T @ A[S].T[U].T + A[M].T[U].T,
+            (A[M].T[M].T + T.T @ A[S].T[M].T + A[M].T[S].T @ T
+             + T.T @ A[S].T[S].T @ T),
+        ]], 'csr')
+    y = np.concatenate((b[U] - A[U].T[S].T @ g,
+                        b[M] - A[M].T[S].T @ g))
+
+    return (
+        B,
+        y,
+        np.zeros_like(b),
```

Comment:
```suggestion
        np.zeros_like(b, dtype=B.dtype),
```
would be great to also add here support for complex values.